assets/prometheus-k8s/service-monitor-thanos-sidecar.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: monitoring.coreos.com/v1
```

Comment:
I would like to deploy this unless any objections? cc @simonpasquier

Reply:
What's the purpose? We don't deploy sidecar, so why deploy ServiceMonitor for it?

Reply:
What do you mean by we don't deploy it, can you clarify? https://github.com/openshift/cluster-monitoring-operator/blob/05808129786cfb4036dc68c63bcc742e76536c4a/assets/prometheus-user-workload/prometheus.yaml#L67

https://github.com/openshift/cluster-monitoring-operator/blob/05808129786cfb4036dc68c63bcc742e76536c4a/assets/prometheus-k8s/prometheus.yaml#L123

And

Reply:
Sorry, I was mistaken. Not enough coffee :smile: 

We should deploy it then :+1:

Reply:
Will Prometheus be able to scrape it? 
We have https://issues.redhat.com/browse/MON-1064 tracking this already.

Reply:
Yes exactly this is why I pinged you, I remember you looked into it. I thought you added it in upstream?

Reply:
I've just checked on a running cluster and the sidecar listens on `127.0.0.1:10902` so it won't work out of the box, we need `kube-rbac-proxy` (or alike) in front.

Reply:
Did you get a chance to look into this at the time and is it straightforward? I suspect it might take some time, so maybe better to leave out of this PR but straight away start work on it? WDYT @paulfantom @simonpasquier @dgrisonnet 

My only concern if we do it in this PR and we then need something urgent to fix a bug this would block it, but happy to also just do that :D

Reply:
> I suspect it might take some time, so maybe better to leave out of this PR but straight away start work on it?

This sounds like a good approach to me.